caffe-distri/src/main/cpp/util/rdma_sync.cpp

```diff
@@ -89,13 +96,20 @@ void RDMASync<Dtype>::CreateWorkerBuffers(int peer) {
   // Send diff from local (rank_) to remote (peer)
   uint8_t* diff = reinterpret_cast<uint8_t*>(diff_ + offs);
   diff_send_[peer].reset(new RDMABuffer(channel, diff, size));
+
+  // Recv ctrl from remote (peer) to local (rank_)
+  uint8_t* ctrl;
+  CUDA_CHECK(cudaMalloc(&ctrl, CTRL_BUF_SIZE));
```

Comment:
A cluster usually consists of a handful nodes, typically 2~8 nodes, could go as high as 16, but it is rare in our application. Sync happens every iteration, basically each node computes its network, waits for all nodes to finish this iteration, exchanges metrics with other nodes, then continues the next iteration.